language/functional-tests/src/evaluator.rs

```diff
@@ -558,12 +558,12 @@ pub fn eval<TComp: Compiler>(
 
     // Set up a fake executor with the genesis block and create the accounts.
     let mut exec = if config.validator_set.payload().is_empty() {
-        FakeExecutor::from_fresh_genesis()
+        FakeExecutor::from_genesis_file()
     } else {
         // use custom validator set. this requires dynamically generating a new genesis tx and
         // is thus more expensive.
         FakeExecutor::custom_genesis(
-            stdlib_modules(StdLibOptions::Fresh).to_vec(),
+            stdlib_modules().to_vec(),
```

Comment:
This results in tests with custom validator configs running with fresh stdlib and others with staged. This is tricker to handle than I initially thought and there are a few thing we need to pay attention to at the same time:
1) Move tests use fresh/fresh + custom validators
2) IR tests use staged/staged + custom validators
3) The genesis transaction is cached across tests, at least for all the regular IR tests.